dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/OracleStatsConnector.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022-2023 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.oracle;
+
+import com.google.edwmigration.dumper.application.dumper.ConnectorArguments;
+import com.google.edwmigration.dumper.application.dumper.task.DumpMetadataTask;
+import com.google.edwmigration.dumper.application.dumper.task.Task;
+import com.google.edwmigration.dumper.plugin.ext.jdk.annotation.Description;
+import java.util.List;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+// TODO: add @AutoService when implementation is completed
+@Description("Dumps aggregated statistics from Oracle")
+@ParametersAreNonnullByDefault
+public class OracleStatsConnector extends AbstractOracleConnector {
+
+  private final StatsTaskListGenerator taskListGenerator = new StatsTaskListGenerator();
```

Comment:
It's hard to tell now, when the implementation is not yet ready, but it seems to me that the lifetime of the generator should be bound by the `addTasksTo` method. Unless you already have code that uses it outside of this method, I would suggest moving it into a local variable. It could always be moved to a field, if the scope is broadened.

Reply:
Done